src/clusterfuzz/_internal/bot/tasks/cron/helpers/bot_manager.py

```diff
@@ -0,0 +1,318 @@
+
+    while True:
+      response = self.execute(
+          self.compute.instanceGroupManagers().listManagedInstances(
+              project=self.project_id,
+              zone=self.zone,
+              instanceGroupManager=self.name,
+              pageToken=next_page_token,
+              filter=instance_filter),
+          result_proc=self._identity)
+
+      for instance in response['managedInstances']:
+        if instance['currentAction'] != 'DELETING':
+          # Instances can be stuck in DELETING, don't include them.
+          yield instance
+
+      if 'nextPageToken' in response:
+        next_page_token = response['nextPageToken']
+      else:
+        break
+
+  def create(self,
+             base_instance_name,
+             instance_template,
+             size=0,
+             auto_healing_policy=None,
+             wait_for_instances=True):
+    """Create this instance group."""
+    manager_body = {
+        'baseInstanceName': base_instance_name,
+        'instanceTemplate': 'global/instanceTemplates/' + instance_template,
+        'name': self.name,
+        'targetSize': size,
+    }
+    if auto_healing_policy:
+      manager_body['autoHealingPolicies'] = [auto_healing_policy]
+
+    result_proc = None
+    if wait_for_instances:
+      result_proc = self._handle_size_change
+
+    self.execute(
+        self.compute.instanceGroupManagers().insert(
+            project=self.project_id, zone=self.zone, body=manager_body),
+        result_proc=result_proc)
+
+  def patch_auto_healing_policies(self,
+                                  auto_healing_policy=None,
+                                  wait_for_instances=True):
+    """Update the health check url of this instance group."""
+    if auto_healing_policy:
+      request_body = {'autoHealingPolicies': [auto_healing_policy]}
+    else:
+      request_body = {'autoHealingPolicies': []}
+
+    result_proc = None
+    if wait_for_instances:
+      result_proc = self._handle_size_change
+
+    self.execute(
+        self.compute.instanceGroupManagers().patch(
+            project=self.project_id,
+            zone=self.zone,
+            instanceGroupManager=self.name,
+            body=request_body),
+        result_proc=result_proc)
+
+  def resize(self, new_size, wait_for_instances=True):
+    """Resize this instance group."""
+    result_proc = None
+    if wait_for_instances:
+      result_proc = self._handle_size_change
+
+    self.execute(
+        self.compute.instanceGroupManagers().resize(
+            project=self.project_id,
+            zone=self.zone,
+            instanceGroupManager=self.name,
+            size=new_size),
+        result_proc=result_proc)
+
+  def delete(self):
+    """Delete this instance group."""
+    self.execute(self.compute.instanceGroupManagers().delete(
+        project=self.project_id, zone=self.zone,
+        instanceGroupManager=self.name))
+
+
+class InstanceTemplate(Resource):
+  """Instance template."""
```

Comment:
Maybe describe what an instance is in this context, the comment isn't so helpful.